### PR TITLE
Feature: uniform steady body force field

### PR DESCRIPTION
**Pull request by TzuYaoHuang**
- gravitational force for example
- tested with 2D circle case: Success and won't mess up with Dirichlet BC.

### CODE REVIEW

File: src/Flow.jl [hunk 1]
```diff
@@ -32,7 +32,7 @@ function median(a,b,c)
     return a
 end
 
-function conv_diff!(r,u,Φ;ν=0.1)
+function conv_diff!(r,u,Φ;ν=0.1,g=(0,0,0,0))
```
Comment (weymouth) [inline src/Flow.jl:35]: I'm very confused by the default value of (0,0,0,0)

File: src/WaterLily.jl [hunk 2]
```diff
@@ -55,9 +55,9 @@ struct Simulation
     pois :: AbstractPoisson
     function Simulation(dims::NTuple{N}, u_BC::NTuple{N}, L::Number;
                         Δt=0.25, ν=0., U=√sum(abs2,u_BC), ϵ=1,
-                        uλ::Function=(i,x)->u_BC[i],exitBC=false,
+                        uλ::Function=(i,x)->u_BC[i],exitBC=false, grav=ntuple(i->0,N),
```
Comment (weymouth) [inline src/WaterLily.jl:58]: I think this is the easiest default

File: src/Flow.jl [hunk 3]
```diff
@@ -71,7 +74,8 @@ struct Flow{D, T, Sf<:AbstractArray{T}, Vf<:AbstractArray{T}, Tf<:AbstractArray{
     Δt:: Vector{T} # time step (stored in CPU memory)
     ν :: T # kinematic viscosity
     exitBC :: Bool # Convection exit
-    function Flow(N::NTuple{D}, U::NTuple{D}; f=Array, Δt=0.25, ν=0., uλ::Function=(i, x) -> 0., T=Float64, exitBC = false) where D
+    g :: NTuple{D, T} # gravity field
+    function Flow(N::NTuple{D}, U::NTuple{D}; f=Array, Δt=0.25, ν=0., uλ::Function=(i, x) -> 0., T=Float64, exitBC = false,g=ntuple(x->zero(T),D)) where D
```
Comment (weymouth) [inline src/Flow.jl:78]: This is ok, but doesn't match below.

**Review by weymouth — COMMENTED** [review]
- Needs some work on the defaults.
- Needs a unit test.
- Can this test a time-varying a? That's something Marin and I will need in the future for accelerating reference frames.

**Comment by TzuYaoHuang** [comment]
I will need to open a new brach since the main branch changes quite a bit. Will fix the problem and reopen the pull request.

**Pull request closed without merging**